### PR TITLE
Removing placeholders. Add StaticHuffmanCode implementation.

### LINKED ISSUE #4: Add Huffman code implementation.
Add the first implementation and tests for Huffman code to the project.

### PR DESCRIPTION
**Pull request by mhco0**
- Closes #4.
- Removes Tests Placeholders.
- Adds the first Implementation and tests for Static Huffman encoding and decoding.
- Adds CXX 20 standard.
- Changes project name to only `doht`.

### CODE REVIEW

**Comment by mhco0** [comment]
@joseviccruz  Please take a look at this MR.

File: include/doht/shc/shc.h [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+class StaticHuffmanCode final {
+ public:
+  explicit StaticHuffmanCode();
+  ~StaticHuffmanCode();
+  StaticHuffmanCode(const StaticHuffmanCode& copy) = delete;
+  StaticHuffmanCode(StaticHuffmanCode&& rval) noexcept = default;
+
+  StaticHuffmanCode operator=(const StaticHuffmanCode&) noexcept = delete;
+
+  std::vector<std::byte> Encode(const std::span<char>& text);
```
Comment (joseviccruz) [inline include/doht/shc/shc.h:21]: Why not `std::string_view`?
Also, std::span must be passed-by-value, see: [link](https://brevzin.github.io/c++/2018/12/03/span-best-span/)

File: src/shc.cpp [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    SetSize(frequencies.size());
+
+    struct HeapNode : Node {
+      HeapNode(const std::optional<char>& symbol, size_t freq) {
+        this->symbol = symbol;
+        this->frequency = freq;
+      }
+
+      std::shared_ptr<HeapNode> left{};
+      std::shared_ptr<HeapNode> right{};
```
Comment (joseviccruz) [inline src/shc.cpp:45]: Couldn't it be std::unique_ptr? I do not see any copies here and unique pointers do not have the same overhead as shared pointers

File: tests/shc.test.cpp [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+TEST_F(StaticHuffmanCodeTest, TestEncodeDecode) {
+  auto encoded = shc_.Encode(text_);
+  auto decoded = shc_.Decode(encoded);
+  ASSERT_EQ(decoded.size(), 15);
+  ASSERT_EQ(decoded, text_);
+}
+
+TEST_F(StaticHuffmanCodeTest, FunnyCharmander) {
+  std::string charmander =
```
Comment (joseviccruz) [inline tests/shc.test.cpp:36]: `std::string_view`?

File: src/shc.cpp [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+
+    std::vector<std::shared_ptr<HeapNode>> leafs{};
+    leafs.reserve(frequencies.size());
+    for (auto& it : frequencies) {
+      leafs.emplace_back(frequency_to_node(it));
+    }
+
+    std::priority_queue<std::shared_ptr<HeapNode>,
+                        std::vector<std::shared_ptr<HeapNode>>,
+                        SharedHeapNodeFunctor>
```
Comment (joseviccruz) [inline src/shc.cpp:82]: Since there is only one occurrence, perhaps you could use std::function and pass a lambda in the constructor of the priority queue

**Review by joseviccruz — COMMENTED** [review]
Main thing: I think I would try to use unique pointers in the occurrences of shared pointers. Also, I would prefer to use string views instead of span<char>

**Pull request merged**